enhancements/api-review/custom-tags-aws.md

```diff
@@ -31,113 +36,325 @@ have the ability to apply user defined tags to many resources created by OpenShi
+   Existing tag for AWS resource = `key_infra1 = value_old`\
+   New tag request = `.spec.platformSpec.aws.resourceTags` has `key_infra1 = value_update1`\
+   Action = Existing tag for AWS resource is updated to reflect new value.\
+   Final tag set to AWS resource = `key_infra1 = value_update1`\
+   Event action = An event is generated to notify user about the action status (success/failure) to update tags for the AWS resource.
+
+3) The new user-defined tag request has empty string `""`, the user-defined tag value for the AWS resource will be updated. Please refer to the `Delete Scenarios` for more details on empty value string handling.\
+   For example,\
+   Existing tag for AWS resource = `key_infra1 = value_old`\
+   New tag request = `.spec.platformSpec.aws.resourceTags` has `key_infra1 =`\
+   Action = Existing tag for AWS resource is updated to reflect new value.\
+   Final tag set to AWS resource = `key_infra1 = value_update1`\
+   Event action = An event is generated to notify user about the action status (success/failure) to update tags for the AWS resource. A warning also must be generated about user-defined tag being marked for deletion.
+
+### Delete tags scenarios
+User-defined tags are deleted when the user sets the user-defined tag value to empty string in `.spec.platformSpec.aws.resourceTags`.
+Also refer to `Precedence` scenario where delete of user-defined tag is not allowed to delete.\
+For example,\
+Existing tag for AWS resource = `key_infra1 = value_old`\
+New tag request = `.spec.platformSpec.aws.resourceTags` has `key_infra1 =`\
+Action = Existing tag for AWS resource is deleted.\
+Final tag set to AWS resource = deleted\
+Event action = An event is generated to notify user about the action status (success/failure) to delete tags for the AWS resource.
+
+### Precedence scenarios
+1) User-defined tags on local objects MUST continue to take precedence during create and update. User-defined tags found on local objects must not be deleted by methods described in `Delete tags scenarios`.
+   A warning is generated to inform the user about action not being applied on the list of user tags.
+
+   For example,\
+   Existing tag in local object like machine CRD = `key_infra1 = custom_value`\
+   New tag request = `.spec.platformSpec.aws.resourceTags` has `key_infra1 = value1`\
+   Action = The tag for AWS resource is maintained to `key_infra1 = custom_value`.\
+   Final tag set to AWS resource = `key_infra1 = value_update1`\
+   Event action = An event is generated to notify user about the action status (success/failure) to update tags for the AWS resource. A warning also must be generated about list of user-defined tags on which action is not applicable.
+
+2) `.spec.platformSpec.aws` take precedence over `.status.platformStatus.aws`. User-defined tags must be merged from `.status.platformStatus.aws` and set to AWS resource.
+   On delete, user-defined must also be removed from `.status.platformStatus.aws`, if present.
```

Comment:
On delete of what? Be clear and specific.

Reply:
Rephrased it, thanks for locating it.